src/praisonai/praisonai/integrations/managed_local.py

```
@@ -451,21 +463,89 @@ def _restore_state(self) -> None:
+                if loop.is_running():
+                    # We're in async context, need to run in executor
+                    import threading
+                    result = [None]
+                    exception = [None]
+                    
+                    def run_install():
+                        try:
+                            new_loop = asyncio.new_event_loop()
+                            asyncio.set_event_loop(new_loop)
+                            result[0] = new_loop.run_until_complete(self._install_via_compute(pip_pkgs))
+                        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)